### PR TITLE
Fix some issues from #613

### PR DESCRIPTION
**Pull request by mmlb**
## Description

#613 introduced a few unwanted bits and a typo.

## Why is this needed

#613 did not use tools.go setup we have for using pinned go tools.

I reworked it so it also didn't do recursive make as that is considered harmful and setup better dependencies so that we let make be better equipped to do the right thing. This way we avoid using make as a dumb command runner, which if not done can lead to racy/broken makefiles (see hook for example).

I fixed a typo that would cause the manager to use tink-server image, likely to not work :D.

In the process I noticed that we don't have `generate-manifests` hooked up to be run as part of ci to detect drift, so I added it as a dependency of `generated` as it should have been from the beginning.

## How Has This Been Tested?

I ran make a bunch.

### CODE REVIEW

File: kube.mk [hunk 1]
```diff
@@ -35,59 +11,55 @@ generate-go: bin/controller-gen bin/gofumpt # Generate Go code.
 	gofumpt -w -s ./pkg/apis
 
 .PHONY: generate-manifests
-generate-manifests: bin/controller-gen # Generate manifests e.g. CRD, RBAC etc.
+generate-manifests: generate-crds generate-rbacs generate-server-rbacs # Generate manifests e.g. CRD, RBAC etc.
```
Comment (micahhausler) [inline kube.mk:14]: want this to be a doc?
```suggestion
generate-manifests: generate-crds generate-rbacs generate-server-rbacs ## Generate manifests e.g. CRD, RBAC etc.
```

Comment (mmlb) [reply kube.mk:14]: Hmm good catch this was from before I just tacked it onto `generate`.

Comment (mmlb) [reply kube.mk:14]: Oh wait I misread the diff. No I don't think its necessary, `make generate` will do the right thing. If you'd like it to be documented I have no problem with it showing up in make help. Do *you* want this to be a doc?

Comment (micahhausler) [reply kube.mk:14]: Ha It doesn't matter to me. Since this isn't a top level target, I'm fine with keeping it not doc'd

**Review by micahhausler — APPROVED** [review]

File: kube.mk (hunk 1 above)
Comment (mmlb) [reply kube.mk:14]: Yeah thats been my thinking.

**Pull request merged**